### PR TITLE
style: remove cyclic dependency between primitives & vm-runner

### LINKED ISSUE #10339: Crate publishing fails
The latest attempt to release crates at 0.18.0 [failed](https://github.com/near/nearcore/actions/runs/7208170912/job/19636528204#step:4:251).

This is due to a circular dependency between near-vm-runner and near-primitives:
- near-primitives depends on near-vm-runner
- near-vm-runner dev-depends on near-primitives

Fixing this would also improve our stance with regards to limited replayability, as it would make near-vm-runner more self-contained.

My initial investigation seems to tell me that we would need to pull the runtime-config-related crates of near-primitives into near-vm-runner, which should hopefully be enough to untangle the circular dependency by removing the dev-dependency.

### PR DESCRIPTION
**Pull request by nagisa**
Everything related to parameters and configurations has been moved into a low-dependency `near-parameters` crate. Thus now any tests that depend on the parameters can do so easily without introducing dependencies on crates that may depend on the crate being tested in the first place (such as `near-vm-runner`.)

One thing of note is that some VM structures pertinent to configuration were moved out of `near-vm-runner` into `near-parameters` as well, so that `near-parameters` is largely a self-contained crate for all things protocol config.

Fixes #10339

### CODE REVIEW

File: core/primitives-core/src/config.rs
Comment (nagisa) [inline core/primitives-core/src/config.rs:2]: Curiously the remaining things in this file did not require movement... :shrug:

File: core/primitives/src/runtime/snapshots/near_primitives__runtime__config_store__tests__139.json.snap
Comment (nagisa) [inline core/primitives/src/runtime/snapshots/near_primitives__runtime__config_store__tests__139.json.snap:1]: Test that originally produced this snapshot is now ignored on nightly.

**Comment by Ekleog-NEAR** [comment]
I’m currently completing my review of https://github.com/near/NEPs/pull/488; I’ll review this either tomorrow or the day after, depending on when I complete that other review :)

**Comment by nagisa** [comment]
Looks like the CI has passed @Ekleog-NEAR

**Comment by Ekleog-NEAR** [comment]
> One thing of note is that some VM structures pertinent to configuration were moved out of `near-vm-runner` into `near-parameters` as well, so that `near-vm-runner` is largely a self-contained crate for all things protocol config.

I think you mean near-parameters here instead of the last near-vm-runner, right?

File: runtime/near-vm-runner/src/tests.rs
Comment (Ekleog-NEAR) [inline runtime/near-vm-runner/src/tests.rs:25]: I’m curious, why did you need this? It seems to not have been needed before

**Review by Ekleog-NEAR — APPROVED** [review]
LGTM! I reviewed assuming that you only:
1. Moved code around
2. Adjusted the paths accordingly
3. Did the test_vm_config change

If you did another meaningful change, please let me know.

I’ll just say that before landing, you should probably fix the top comment, replacing near-vm-runner with near-parameters, for the squashed commit message to be correct :)

File: runtime/near-vm-runner/src/tests.rs
Comment (nagisa) [reply runtime/near-vm-runner/src/tests.rs:25]: It (the call to `replace_with_wasmtime_if_unsupported`) is there in the previous version, though?

Comment (Ekleog-NEAR) [reply runtime/near-vm-runner/src/tests.rs:25]: Oh I had just seen the match and missed it just below, sorry 😅

**Pull request merged**
